### PR TITLE
dev: removed unnecessary @ts-expect-errors

### PR DESCRIPTION
**Pull request by rjchow**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This is part of https://github.com/woocommerce/woocommerce/issues/53939

When the dependencies were upgraded to wp-6.6, most of the @ts-expect-errors that were put in place due to lacking types were made unnecessary and became type checking errors.

This PR removes the ones that can be removed - notably the packages that still don't have types exported as of wp-6.6 are:
- @wordpress/blocks
- @wordpress/block-editor
- @wordpress/media-utils
- @wordpress/notices -- this isn't really missing types it's just that @types/wordpress__notices has a bigger set of exported types.
- @wordpress/edit-site -- this doesn't have types in DefinitelyTyped either

Also notably, there were a bunch of @ts-ignores . These **did not** turn into errors automatically, and I've removed the ones that I noticed. I know that there's still a ton more and I will make a follow up issue to address these.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Inspect the code and observe that no actual code changes were made
2. Observe that build is fine
3. Observe that I have removed TS2578 from the warn-conversion and the build does not report any failure 

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/client/admin/client/customize-store/assembler-hub/sidebar/pattern-screen/sidebar-pattern-screen.tsx [hunk 1]
```diff
@@ -10,18 +10,14 @@ import { __ } from '@wordpress/i18n';
...
 // eslint-disable-next-line @woocommerce/dependency-group
 import { useIsSiteEditorLoading } from '@wordpress/edit-site/build-module/components/layout/hooks';
 // eslint-disable-next-line @woocommerce/dependency-group
 import {
-	store as coreStore,
-	// @ts-expect-error No types for this exist yet.
-} from '@wordpress/core-data';
-// eslint-disable-next-line @woocommerce/dependency-group
-import {
+	// @ts-expect-error No types forthis exist yet.
```
Comment (ilyasfoo) [inline plugins/woocommerce/client/admin/client/customize-store/assembler-hub/sidebar/pattern-screen/sidebar-pattern-screen.tsx]:
```suggestion
	// @ts-expect-error No types for this exist yet.
```

File: plugins/woocommerce/client/admin/client/task-lists/fills/steps/location.tsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { COUNTRIES_STORE_NAME } from '@woocommerce/data';
 import { Fragment, useState } from '@wordpress/element';
 import { Form, FormContextType, Spinner } from '@woocommerce/components';
 import { useSelect } from '@wordpress/data';
-import { Status, Options } from 'wordpress__notices';
+import type { Status, Options } from 'wordpress__notices';
```
Comment (ilyasfoo) [inline plugins/woocommerce/client/admin/client/task-lists/fills/steps/location.tsx:10]: Super nit
```suggestion
import type { Status, Options } from 'wordpress__notices';

```

**Review by ilyasfoo — APPROVED** [review]
Thanks for addressing this, should be a good base for others to start working on fixing types! Code looks good. I also did quick smoke test on core profiler and homescreen and saw nothing unexpected.

> I know that there's still a ton more and I will make a follow up issue to address these.

I propose using Cursor's composer with agent mode to scour through the codebase, I believe it's a good use case.

Left a few nit comments, pre-approving! 🚀

**Review by chihsuan — APPROVED** [review]
Nice! Looks good to me. 💯

**Pull request merged**